### PR TITLE
[moe] support openmoe train

### PR DESCRIPTION
**Pull request by oahzxl**
* support openmoe train for zero2 and zero2+ep
* rewrite checkpoint for ckpt load and save
* use chunk and checkpoint to reduce lm_head's high activation memory due to large vocab size
* update model details to adapt to train

TODO: align result and accuracy

### CODE REVIEW

File: colossalai/nn/layer/moe/checkpoint.py [hunk 1]
```diff
@@ -7,7 +8,7 @@
 from torch.optim import Optimizer
 
 from colossalai.checkpoint_io import CheckpointIO
-from colossalai.tensor.moe_tensor.api import get_ep_group
+from colossalai.tensor.moe_tensor.api import get_dp_rank, get_ep_group, get_ep_rank, get_ep_size, is_moe_tensor
 
 
 class MoeCheckpintIO(CheckpointIO):
```
Comment (ver217) [inline colossalai/nn/layer/moe/checkpoint.py:14]: Maybe you should use `GeneralCheckpointIO`.

Comment (oahzxl) [reply colossalai/nn/layer/moe/checkpoint.py]: done

File: colossalai/context/moe_context.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ def __init__(self):
         self.max_ep_size = None
         self.min_dp_size = None
```
Comment (ver217) [inline colossalai/context/moe_context.py:22]: This class does not have `__enter__` or `__exit__` method. It seems that it's not a context manager. By the way, why is it asingleton?

Comment (oahzxl) [reply colossalai/context/moe_context.py:22]: it is just a gloabl value to store some settings and is designed by haichen

Comment (ver217) [reply colossalai/context/moe_context.py:22]: Can you move it to `colossalai/moe`? And `colossalai/nn/layer/moe` can move there too.

Comment (oahzxl) [reply colossalai/context/moe_context.py:22]: ok. i will do it in the following pr

**Review by ver217 — APPROVED** [review]
